cmd/fs/helpers.go

```diff
@@ -46,6 +48,65 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(pathArgCount int, onlyDirs bool, filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error)) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClient(cmd, args) // TODO: Fix this
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isValidPrefix := isDbfsPath(toComplete)
+
+		if !isValidPrefix {
+			return []string{dbfsPrefix}, cobra.ShellCompDirectiveNoSpace
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		wsc := root.WorkspaceClient(cmd.Context())
+		_, err = wsc.CurrentUser.Me(cmd.Context())
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		completer := completer.NewCompleter(cmd.Context(), filer, onlyDirs)
+
+		if len(args) < pathArgCount {
```

Comment:
You could inverse condition and avoid additional nesting

Reply:
Nice, fixed!